src/OpenColorIO/GpuShaderDesc.cpp

```diff
@@ -294,6 +296,103 @@ void GpuShaderCreator::createShaderText(const char * shaderDeclarations,
+        kw.newLine() << "vector4 __operator__mul__(vector4 v, matrix m)";
+        kw.newLine() << "{";
+        kw.indent();
+        kw.newLine() << "return vector4(v.x * m[0][0] + v.y * m[1][0] + v.z * m[2][0] + v.w * m[3][0], "\
+                                       "v.x * m[0][1] + v.y * m[1][1] + v.z * m[2][1] + v.w * m[3][1], "\
+                                       "v.x * m[0][2] + v.y * m[1][2] + v.z * m[2][2] + v.w * m[3][2], "\
+                                       "v.x * m[0][3] + v.y * m[1][3] + v.z * m[2][3] + v.w * m[3][3]);";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 __operator__mul__(color4 c, vector4 v)";
+        kw.newLine() << "{";
+        kw.indent();
+        kw.newLine() << "return vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a) * v;";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 __operator__mul__(vector4 v, color4 c)";
+        kw.newLine() << "{";
+        kw.indent();
+        kw.newLine() << "return v * vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a);";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 __operator__sub__(color4 c, vector4 v)";
+        kw.newLine() << "{";
+        kw.indent();
+        kw.newLine() << "return vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a) - v;";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 __operator__add__(vector4 v, color4 c) {";
+        kw.indent();
+        kw.newLine() << "return v + vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a);";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 __operator__add__(color4 c, vector4 v) {";
+        kw.indent();
+        kw.newLine() << "return vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a) + v;";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 pow(color4 c, vector4 v) {";
+        kw.indent();
+        kw.newLine() << "return pow(vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a), v);";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "vector4 max(vector4 v, color4 c) {";
+        kw.indent();
+        kw.newLine() << "return max(v, vector4(c.rgb.r, c.rgb.g, c.rgb.b, c.a));";
+        kw.dedent();
+        kw.newLine() << "}";
+
+        kw.newLine() << "";
+        kw.newLine() << "/* The shader implementation */";
+        kw.newLine() << "";
+        kw.newLine() << "shader " << "OSL_" << getFunctionName() 
+                     << "(color4 inColor = {color(0), 1}, output color4 outColor = {color(0), 1})";
+        kw.newLine() << "{";
+
+        const std::string str = kw.string() + getImpl()->m_declarations;
+        getImpl()->m_declarations = str;
+    }
+
+    if (getLanguage() == LANGUAGE_OSL)
+    {
```

Comment:
If I'm reading this correctly, you end the "if OSL" block and then immediately re-enter a new one?